subsys/net/ip/tcp2.c

```diff
@@ -832,6 +940,45 @@ static struct tcp *tcp_conn_new(struct net_pkt *pkt)
 	return conn;
 }
 
+static void update_acked_data(struct tcp *conn, struct tcphdr *th)
+{
+	if (!th) {
+		return;
+	}
+
+	if (!net_tcp_seq_greater(th_ack(th), conn->seq)) {
```

Comment:
Shall we use a simpler check 'conn->seq >= th_ack(th)' here?

Reply:
I am not sure how we could do that, the question is how would we catch the u32_t wraps then.